app/presenters/confirmation_email_addresses_presenter.rb

```diff
@@ -1,8 +1,12 @@
-# TODO: take a look at this rubocop warning
-# rubocop:disable Style/StructInheritance
-class ConfirmationEmailAddressesPresenter < Struct.new(:claim)
+class ConfirmationEmailAddressesPresenter
   delegate :primary_claimant, :representative, :secondary_claimants, to: :claim
 
+  attr_reader :claim
+
+  def initialize(claim)
+    @claim = claim
```

Comment:
I always see this as 'internal knowledge' of how attr_reader etc.. works - ie that it reads an instance variable and it writes to an instance variable.

I would normally do attr_writer :claim in the private section of the class, and call self.claim = claim in the initializer

The advantage is then that everything that changes stuff goes through a writer method - all in one place so easier to debug etc..

Reply:
@garytaylor can you please elaborate a bit more? I can't see how a writer method can make it easier to debug than a plain variable assignment.

Reply:
@garytaylor hmm that is an interesting way. I personaly never seen this approach. This implementation for me is a standard ruby way how to do this. @leoapost what do you think?

Reply:
I agree with @zaparka and I don't see any pros which would worth having private writer methods for instance variables.

I am happy to hear @garytaylor's approach regardless :)

Reply:
@leoapost  - with a writer method - its 1 single place where the value gets written - not multiple.  So , say if you had a bug where something somewhere was changing this value.  With a writer method - simply override the default method, put a pry debug point in or whatever you use and you can see what is making the changes.  Without the writer method - do a search through the code, looking for anything that changes the instance var and put a pry debug point in there.

Its just the way ive always done it because I find it easier - but there isn't a standard ruby way of doing it - there are several !!